Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift

```diff
@@ -129,38 +132,123 @@ public class DefaultAbly: AblyCommon {
+                }
+
+                try disconnectChannel(channel: channel, presenceData: presenceData)
+            })
+            
+            channels.removeValue(forKey: trackableId)
+            completion(.success(true))
+        }catch AblyError.connectionError(let errorInfo){
+            completion(.failure(errorInfo.toErrorInformation()))
+        }catch{
+            completion(.failure(ErrorInformation(error: AblyError.connectionError(errorInfo: ARTErrorInfo.create(withCode: 100_000, message: "Unknown error while disconnecting")))))
+        }
+    }
+    
+    private func disconnectChannel(channel:AblySDKRealtimeChannel, presenceData: PresenceData) throws{
+        leavePresence(channel: channel, presenceData: presenceData) { leaveError in
+            if let leaveError = leaveError {
+                throw AblyError.connectionError(errorInfo: leaveError)
+            }
+            
+            channel.unsubscribe()
+            channel.presence.unsubscribe()
+            self.detachFromChannel(channel: channel) { detachError in
+                guard let error = detachError else { return }
+                throw AblyError.connectionError(errorInfo: error)
+            }
+        }
+    }
+    
+    private func leavePresence(channel:AblySDKRealtimeChannel, presenceData: PresenceData?, completion : @escaping (ARTErrorInfo?) throws -> Void){
         let presenceDataJSON: Any?
         if let presenceData = presenceData {
             presenceDataJSON = self.presenceDataJSON(data: presenceData)
         } else {
             presenceDataJSON = nil
         }
         
-        channelToRemove.presence.leave(presenceDataJSON) { [weak self] error in
-            guard let error = error else {
-                self?.logHandler?.d(message: "\(String(describing: Self.self)): Left channel [id: \(trackableId)] presence successfully", error: nil)
-                channelToRemove.presence.unsubscribe()
-                channelToRemove.unsubscribe()
-                
-                channelToRemove.detach { [weak self] detachError in
-                    guard let error = detachError else {
-                        self?.channels.removeValue(forKey: trackableId)
-                        completion(.success(true))
-                        
-                        return
-                    }
-                    self?.logHandler?.e(message: "\(String(describing: Self.self)): Error during detach channel [id: \(trackableId)] presence", error: error)
-                    completion(.failure(error.toErrorInformation()))
-                }
-                
-                return
+         channel.presence.leave(presenceDataJSON){ errorInfo in
+             //This is to maintain closure signature
+             do{
+                 try completion(errorInfo)
+             }catch{}
+        }
+
+    }
+    
+    private func detachFromChannel(channel:AblySDKRealtimeChannel, completion : @escaping (ARTErrorInfo?) throws -> Void) {
```

Comment:
As in my comment on `leavePresence`, I don't think that `completion` should have the `throws` modifier.

Reply:
This will have been addressed https://github.com/ably/ably-asset-tracking-swift/commit/248ff80d6b48c0c988fa637c81c01c9ded685831